pkg/oc/cli/newapp/newapp.go

```diff
@@ -443,71 +444,81 @@ func (o *AppOptions) RunNewApp() error {
 		return nil
 	}
 
+	supportedTypes := map[schema.GroupVersionKind]bool{
+		{Version: "v1", Kind: "Pod"}:                                    true,
+		{Group: buildapi.GroupName, Version: "v1", Kind: "BuildConfig"}: true,
+		{Group: imageapi.GroupName, Version: "v1", Kind: "ImageStream"}: true,
+		{Group: routeapi.GroupName, Version: "v1", Kind: "Route"}:       true,
+	}
+
 	hasMissingRepo := false
 	installing := []*corev1.Pod{}
 	indent := o.Action.DefaultIndent()
 	containsRoute := false
 	for _, item := range result.List.Items {
 		// these are all unstructured
 		unstructuredObj := item.(*unstructured.Unstructured)
-		obj, err := legacyscheme.Scheme.New(unstructuredObj.GroupVersionKind())
-		if err != nil {
-			return err
-		}
-		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredObj.Object, obj); err != nil {
-			return err
-		}
-		switch t := obj.(type) {
-		case *corev1.Pod:
-			if t.Annotations[newcmd.GeneratedForJob] == "true" {
-				installing = append(installing, t)
+
+		// Determine if dealing with a "known" resource, containing a switch case below.
+		// If so, go through with a conversion attempt, and fail if necessary.
+		if supported := supportedTypes[unstructuredObj.GroupVersionKind()]; supported {
```

Comment:
i think this goes back to @deads2k's comment.  if it's `!supported` you can just do a `continue` here, because there's nothing else in the loop after the `if` block.